### PR TITLE
Bump dependencies

### PR DESCRIPTION
**Pull request by MbolotSuse**
Bumps go version to 1.19 and bumps the version of rancher

### CODE REVIEW

**Comment by MbolotSuse** [comment]
Related to https://github.com/rancher/rancher/pull/42062. Necessary so that this component can validate tokens hashed using Rancher's new methods of token hashing.

**Review by macedogm — APPROVED** [review]

**Comment by macedogm** [comment]
@MbolotSuse thanks for making this PR. Since we will have to cut a new -rc for `kube-api-auth` and then propagate it to r/r, would you mind in also reviewing/approving the following PRs, so we do this process only once?

- https://github.com/rancher/kube-api-auth/pull/16 - the base image is EOL
- https://github.com/rancher/kube-api-auth/pull/17 - bump images in Drone to fix the pipeline error (you might have to merge this PR before yours)

Thanks!

File: go.mod [hunk 1]
```diff
@@ -1,236 +1,248 @@
 module github.com/rancher/kube-api-auth
 
-go 1.17
+go 1.19
+
+// note: replace once 1.1.1 has been cut
+replace github.com/rancher/wrangler v1.1.1 => github.com/rancher/wrangler v1.1.1-0.20230705223603-201b4da5bdaf
```
Comment (macedogm) [inline go.mod:6]: Isn't v1.1.1 already cut https://github.com/rancher/wrangler/releases/tag/v1.1.1?

**Comment by macedogm** [comment]
@MbolotSuse another point, would you consider also bumping the following dependencies, please? They solve quite a few CVEs.

```diff
diff --git a/go.mod b/go.mod
index 9f25b1f..3d1a5a1 100644
--- a/go.mod
+++ b/go.mod
@@ -6,9 +6,9 @@ go 1.19
 replace github.com/rancher/wrangler v1.1.1 => github.com/rancher/wrangler v1.1.1-0.20230705223603-201b4da5bdaf
 
 replace (
-       github.com/docker/distribution => github.com/docker/distribution v2.7.1+incompatible // oras dep requires a replace is set
+       github.com/docker/distribution => github.com/docker/distribution v2.8.2+incompatible // oras dep requires a replace is set
 
-       github.com/docker/docker => github.com/docker/docker v20.10.6+incompatible // oras dep requires a replace is set
+       github.com/docker/docker => github.com/docker/docker v20.10.24+incompatible // oras dep requires a replace is set
        github.com/knative/pkg => github.com/rancher/pkg v0.0.0-20190514055449-b30ab9de040e
 
        github.com/rancher/lasso => github.com/rancher/lasso v0.0.0-20230629200414-8a54b32e6792
@@ -27,32 +27,32 @@ replace (
 
        helm.sh/helm/v3 => github.com/rancher/helm/v3 v3.9.0-rancher1
 
-       k8s.io/api => k8s.io/api v0.25.4
-       k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.25.4
-       k8s.io/apimachinery => k8s.io/apimachinery v0.25.4
-       k8s.io/apiserver => k8s.io/apiserver v0.25.4
-       k8s.io/cli-runtime => k8s.io/cli-runtime v0.25.4
+       k8s.io/api => k8s.io/api v0.25.11
+       k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.25.11
+       k8s.io/apimachinery => k8s.io/apimachinery v0.25.11
+       k8s.io/apiserver => k8s.io/apiserver v0.25.11
+       k8s.io/cli-runtime => k8s.io/cli-runtime v0.25.11
        k8s.io/client-go => github.com/rancher/client-go v1.25.4-rancher1
-       k8s.io/cloud-provider => k8s.io/cloud-provider v0.25.4
-       k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.25.4
-       k8s.io/code-generator => k8s.io/code-generator v0.25.4
-       k8s.io/component-base => k8s.io/component-base v0.25.4
-       k8s.io/component-helpers => k8s.io/component-helpers v0.25.4
-       k8s.io/controller-manager => k8s.io/controller-manager v0.25.4
-       k8s.io/cri-api => k8s.io/cri-api v0.25.4
-       k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.25.4
-       k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.25.4
-       k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.25.4
-       k8s.io/kube-proxy => k8s.io/kube-proxy v0.25.4
-       k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.25.4
-       k8s.io/kubectl => k8s.io/kubectl v0.25.4
-       k8s.io/kubelet => k8s.io/kubelet v0.25.4
-       k8s.io/kubernetes => k8s.io/kubernetes v1.25.4
-       k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.25.4
-       k8s.io/metrics => k8s.io/metrics v0.25.4
-       k8s.io/mount-utils => k8s.io/mount-utils v0.25.4
-       k8s.io/pod-security-admission => k8s.io/pod-security-admission v0.25.4
-       k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.25.4
+       k8s.io/cloud-provider => k8s.io/cloud-provider v0.25.11
+       k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.25.11
+       k8s.io/code-generator => k8s.io/code-generator v0.25.11
+       k8s.io/component-base => k8s.io/component-base v0.25.11
+       k8s.io/component-helpers => k8s.io/component-helpers v0.25.11
+       k8s.io/controller-manager => k8s.io/controller-manager v0.25.11
+       k8s.io/cri-api => k8s.io/cri-api v0.25.11
+       k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.25.11
+       k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.25.11
+       k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.25.11
+       k8s.io/kube-proxy => k8s.io/kube-proxy v0.25.11
+       k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.25.11
+       k8s.io/kubectl => k8s.io/kubectl v0.25.11
+       k8s.io/kubelet => k8s.io/kubelet v0.25.11
+       k8s.io/kubernetes => k8s.io/kubernetes v1.25.11
+       k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.25.11
+       k8s.io/metrics => k8s.io/metrics v0.25.11
+       k8s.io/mount-utils => k8s.io/mount-utils v0.25.11
+       k8s.io/pod-security-admission => k8s.io/pod-security-admission v0.25.11
+       k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.25.11
 )
 
 require (
@@ -222,10 +222,10 @@ require (
        k8s.io/api v0.26.1 // indirect
        k8s.io/apiextensions-apiserver v0.25.4 // indirect
        k8s.io/apimachinery v0.26.1 // indirect
-       k8s.io/apiserver v0.25.4 // indirect
-       k8s.io/cli-runtime v0.25.4 // indirect
+       k8s.io/apiserver v0.25.11 // indirect
+       k8s.io/cli-runtime v0.25.11 // indirect
        k8s.io/client-go v12.0.0+incompatible // indirect
-       k8s.io/component-base v0.25.4 // indirect
+       k8s.io/component-base v0.25.11 // indirect
        k8s.io/gengo v0.0.0-20220902162205-c0856e24416d // indirect
        k8s.io/helm v2.16.7+incompatible // indirect
        k8s.io/klog v1.0.0 // indirect
@@ -236,7 +236,7 @@ require (
        k8s.io/kubernetes v1.25.4 // indirect
        k8s.io/utils v0.0.0-20230209194617-a36077c30491 // indirect
        oras.land/oras-go v1.1.0 // indirect
-       sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.0.33 // indirect
+       sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.0.37 // indirect
        sigs.k8s.io/cli-utils v0.27.0 // indirect
        sigs.k8s.io/cluster-api v1.2.12 // indirect
        sigs.k8s.io/controller-runtime v0.12.3 // indirect
```

File: go.mod (hunk 1 above)
Comment (MbolotSuse) [reply go.mod:6]: The comment is in error - the replaced commit is for changes after 1.1.1. We need to replace once 1.1.2 (or whatever the next release of wrangler is) has been cut.

**Comment by MbolotSuse** [comment]
@macedogm I've made the dependency bumps as requested, and also merged #14 and #16 into this PR. LMK if you would like any other changes.

**Review by macedogm — APPROVED** [review]

**Comment by macedogm** [comment]
@MbolotSuse LGTM and thanks for putting all PRs together 👍🏻

**Review by superseb — APPROVED** [review]

**Pull request merged**